yoti_python_sdk/dynamic_sharing_service/share_url.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import json
+
+INVALID_DATA = "Json is incorrect, contains invalid data"
+APPLICATION_NOT_FOUND = "Application was not found"
+UNKNOWN_ERROR = "Unknown HTTP error occured"
+
+
+def create_share_url(yoti_client, dynamic_scenario):
+    http_method = "POST"
+    payload = json.dumps(dynamic_scenario, sort_keys=True).encode()
+    endpoint = yoti_client.endpoints.get_dynamic_share_request_url()
+    response = yoti_client.make_request(http_method, endpoint, payload)
+
+    status_code = response.status_code
+    if status_code >= 200 and status_code < 300:
+        response_json = json.loads(response.text)
+
+        return ShareUrl(
+            _ShareUrl__qr_code=response_json["qrcode"],
+            _ShareUrl__ref_id=response_json["ref_id"],
+        )
+
+    elif status_code == 400:
```

Comment:
Since this logic would also apply to the other requests we make (AML & Profile), could we extract this logic out into a separate class and utilise it both here and in the other requests? (might have to change the 400 error to be more generic)